portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -82,4 +88,33 @@ private String getParameter(HttpServletRequest request, String name, String defa
     }
     return value;
   }
+
+  /**
+   * @return the number of comments to display, or -1 if all comments should be displayed
+   */
+  private int getNumCommentsToDisplay(HttpServletRequest request) {
+    // Get the input from the form.
+    String numCommentsString = getParameter(request, "num-comments", null);
+
+    if (numCommentsString.equals("All")) {
+      return -1;
+    }
+
+    // Convert the input to an int.
+    int numComments;
+    try {
+      numComments = Integer.parseInt(numCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + numCommentsString);
```

Comment:
This is fine as well, but you may consider using a logger over System.err/out which offers more flexibility like log levels. There are a few options like java.util.logging or Log4j.